# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/DeviceManagerDx.cpp
# Commit: d158b20d9a65
# Full Hash: d158b20d9a65cd05dd69a586473216b800523083
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-10-21 09:51:28
# Regressor Bug: 1595994
# File Overlap Count: 1
# Description:
#   Bug 1595994 - PD: Allow WMF decoding inside RDD process. r=mattwoodrow
#   
#   This assumes that access to Windows WMF and DirectX is allowed from RDD process.
#   
#   Depends on D91699
# ==============================================================================

diff -r d0c9c64867cc -r d158b20d9a65 gfx/thebes/DeviceManagerDx.cpp
--- a/gfx/thebes/DeviceManagerDx.cpp	Tue Oct 20 23:26:49 2020 +0000
+++ b/gfx/thebes/DeviceManagerDx.cpp	Tue Oct 20 23:26:52 2020 +0000
@@ -509,8 +509,9 @@
     // should never reach here. Furthermore, the UI process does not create
     // devices when using a GPU process.
     //
-    // So, this should only ever get called on the content process.
-    MOZ_ASSERT(XRE_IsContentProcess());
+    // So, this should only ever get called on the content process or RDD
+    // process
+    MOZ_ASSERT(XRE_IsContentProcess() || XRE_IsRDDProcess());
 
     // In the child process, we search for the adapter that matches the parent
     // process. The first adapter can be mismatched on dual-GPU systems.